src/android/HockeyApp.java

```diff
@@ -27,14 +31,67 @@
 import java.lang.Thread;
 import java.text.SimpleDateFormat;
 import java.util.Calendar;
+import java.util.ArrayList;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.view.View;
 
 public class HockeyApp extends CordovaPlugin {
 
     public static boolean initialized = false;
     public static String appId;
+    public static Object monitor = new Object();
+    public static volatile Bitmap bitmap;
     
     private ConfiguredCrashManagerListener crashListener;
+    
+    // Integration with Crosswalk requires:
+    // - Crosswalk engine version 18 or later
+    // - Latest cordova-plugin-crosswalk-webview plugin     
+    private Bitmap getBitmap() {
+        bitmap = null;
+        boolean isCrosswalk = false;
+        try {
+            Class.forName("org.crosswalk.engine.XWalkWebViewEngine");
+            isCrosswalk = true;
+        } catch (Exception e) {
+        }
+
+        if (isCrosswalk) {
+            long start = System.currentTimeMillis();
+            webView.getPluginManager().postMessage("captureXWalkBitmap", this);
+            try {
+                // Can't take more than 5 seconds to get a screenshot
+                while (bitmap == null && System.currentTimeMillis() - start < 5000) {
+                    synchronized (monitor) {
+                        monitor.wait(50);
```

Comment:
Not sure if commonly an issue, but the [documentation](http://developer.android.com/reference/java/lang/Object.html#wait(long)) of `Object.wait()` says:

> A waiting thread can be sent interrupt() to cause it to prematurely stop waiting, so wait should be called in a loop to check that the condition that has been waited for has been met before continuing.

I will change the math so that at best it only waits once but keep the loop just in case.
I will also create a constant for the max wait time.